Replace embedded SlickGrid with rails-assets gem

Fryguy opened the pull request:
@skateman Please review.

cc @dclarizio @martinpovolny

Fryguy commented:
WIP because I need help to verify this actually works.

One general note: slickgrid does not specify any dependencies in a bower.json file, so we just get what's there.

Fryguy commented on app/assets/javascripts/import.js:
```diff
@@ -1,4 +1,4 @@
-//= require_directory ./SlickGrid-2.1/
+//= require_directory slickgrid
```
Is this needed and is this right since the directory now comes from a gem?

Fryguy commented on app/assets/javascripts/miq_policy/import.js:
```diff
@@ -1,4 +1,4 @@
-//= require_directory ../SlickGrid-2.1/
+//= require_directory slickgrid
```
Same here: Is this needed and is this right since the directory now comes from a gem?

Fryguy commented:
There is also a product_slickgrid.css file.  Is that one ok, or do we need to change that one as well?

skateman commented:
@Fryguy I've backtraced the required files and instead the `require_directory` you should use these:
```javascript
//= require slickgrid/lib/jquery.event.drag-2.2
//= require slickgrid/slick.core
//= require slickgrid/slick.dataview
//= require slickgrid/slick.grid
```

Fryguy commented:
Thanks @skateman I'll change it

Fryguy commented:
@skateman Updated...can you please review?  I'm not sure the best way to verify this works.

skateman commented:
@Fryguy check the page sources, if it includes the same assets, then it should be okay!

Fryguy commented:
@skateman, So I tried with miq_policy/import.js by going to the page it's used from, but I can't find it.  Navigating to Control -> Import/Export takes you to miq_policy/export URL.  I cannot find a way to get to miq_policy/import URL.  If I try going directly to the URL I get the below.  Help?

```
[----] I, [2015-10-02T13:09:31.836539 #9579:3fd071c5e208]  INFO -- : Filter chain halted as :check_privileges rendered or redirected
[----] I, [2015-10-02T13:09:31.836659 #9579:3fd071c5e208]  INFO -- : Completed 403 Forbidden in 20ms (Views: 0.6ms | ActiveRecord: 1.3ms)
```

Fryguy commented:
@skateman All of the other screens have the new js assets.

skateman commented:
@Fryguy log in again? maybe the credentials have expired

Fryguy commented:
nope...I did it immediately after I went to the export screen.

h-kataria commented:
@fryguy, you need to select a file to import on Policy Import/Export screen then press upload button that should bring you to miq_policy/import view

h-kataria commented:
step 1:
![policy_browse_upload](https://cloud.githubusercontent.com/assets/3450808/10351495/d76bf696-6d17-11e5-8bd3-cee9e6b20ed0.png)

step 2:
![policy_import_screen](https://cloud.githubusercontent.com/assets/3450808/10351500/dca05c74-6d17-11e5-907c-1bfab6a7a98f.png)

Fryguy commented:
OK, I've updated the PR with a rebase against #4051.

Along the way I found some really interesting issues, but I think they can be fixed in a follow up PR.

First, the styling seems to have changed a little.

Before:
<img width="1013" alt="screen shot 2015-10-09 at 1 41 51 pm" src="https://cloud.githubusercontent.com/assets/52120/10402288/80fad0bc-6e92-11e5-9e01-81231bcd89be.png">

After:
<img width="1010" alt="screen shot 2015-10-09 at 1 46 37 pm" src="https://cloud.githubusercontent.com/assets/52120/10402290/854fde64-6e92-11e5-920d-6434ac9eb61c.png">

The header seems to have grown compared to before.  But if you notice, the styling was broken even *before* my changes, where the rows are not colored accurately, not to mention the other page elements such as the Submit button (not shown) cc @epwinchell 

---

In trying to understand why the styling may have changed I noticed we had modified the original slick grid files for some reason, which I think was bad on our part.  First is [`slick.grid.js`](https://gist.github.com/Fryguy/a3c2dd1136abbec5788b#file-slick-grid-js-diff).  I'm not sure why these changes were made, but it seems related to IE?

The next file is [`plugins/slick.rowselectionmodel.js`](https://gist.github.com/Fryguy/a3c2dd1136abbec5788b#file-slick-rowselectionmodel-js-diff).  @eclarizio I think you grabbed this from master and not from 2.1.0.  Does that matter for you?

The last file is [`slick.grid.css`](https://gist.github.com/Fryguy/a3c2dd1136abbec5788b#file-slick-grid-css-diff).  In the gem, they use scss, so it's obviously generated.  For comparison, I grabbed the generated version to compare to the one we had committed.  This seems to have affected most of the styling, but I think changes belong in an override file anyway, not changing the originals.  @epwinchell ?

---

@eclarizio @h-kataria @skateman @dclarizio @epwinchell  Can you please verify this PR?  It's up to you how you guys want to proceed, so please let me know.  Personally, I think this can be merged as is, and then styling cleaned up separately.

Fryguy commented:
@epwinchell One more thing to mention about the css file is that the old css file referenced images called `images/sort-desc.gif` and `images/sort-desc.gif`, but they don't exist.  The new location does have the gem's image at `/assets/slickgrid/images/sort-desc.gif` and `/assets/slickgrid/images/sort-desc.gif`, but they don't seem to be working...perhaps they need to be added to the precompile list?

Fryguy commented:
Thanks @epwinchell ...I've updated the product_slickgrid.css with just those diffs from the `slick.grid.css`

epwinchell commented:
@dclarizio tested. looks good
